GVFS/GVFS.Common/Git/GitAuthentication.cs

```diff
@@ -24,12 +24,15 @@ public class GitAuthentication
         private bool isCachedCredentialStringApproved = false;
 
         private bool isInitialized;
+        private ICredentialManager credentialManager;
 
         public GitAuthentication(GitProcess git, string repoUrl)

```

Comment:
At this point we can have this class take a `ICredentialManager` in the constructor as a dependency instead of the `GitProcess` class since it is no longer used.

Reply:
That would be nice - alas, we still have one other call to `GitProcess` in this class:

```
git.TryGetConfigUrlMatch("http", this.repoUrl, out Dictionary<string, GitConfigSetting> configSettings)
```

Reply:
Maybe that call is the responsibility of the creator of this class so that this one becomes:
`public GitAuthentication(ICredentialStore store, Dictionary<string, GitConfigSetting> configSettings)`
or
`public GitAuthentication(ICredentialStore store, GitSsl gitSsl)`